### DISCUSSION TITLE
Custom functional for Relations (Reference)

### DISCUSSION THREAD

**Discussion by nonamich** [prisma/prisma#9837] [Q&A] [answered]
I really like the convenience of prisma.js, I didn't know before that ORM could be so handy. Thanks for this.

I use "[Relations (Reference)](https://www.prisma.io/docs/concepts/components/prisma-schema/relations)" functionality in schema.prisma, it's very convenient but slow because for "[Relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations)" each separate query to database is done, which for me is long, at least 100ms. To reduce response time I had to write custom SQL using [JOIN](https://www.w3schools.com/sql/sql_join_left.asp) and "[Prisma Raw database access](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)" which reduced latency to 40ms.

My question is if I can change "relation" to use [SQL JOIN](https://www.w3schools.com/sql/sql_join_left.asp) or convenient PostgreSQL methods, for example to work with JSON, maybe there is a possibility in prisma.js to write own addons or plugins which could modify or add new "[Relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations)"?

**Comment by ryands17** [comment, marked as answer]
@dimaeromin97 👋 

Fetching relation separately was by design but we have a feature request here so it would be great if you could add a 👍 to this request: https://github.com/prisma/prisma/issues/5184

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
